javaagent-core/src/main/java/org/hypertrace/agent/core/instrumentation/RequestBodyCaptureMethod.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright The Hypertrace Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.hypertrace.agent.core.instrumentation;
+
+/**
+ * Describes instrumentation strategies for capturing the request body on containers. Generally,
+ * this is detected at runtime and stored in the {@link
+ * io.opentelemetry.javaagent.instrumentation.api.InstrumentationContext} so that {@link
+ * #APP_SERVER} instrumentation can short-circuit when it is know that that the preferred {@link
+ * #SERVLET} instrumentation is sufficient to capture the request body
+ */
+public enum RequestBodyCaptureMethod {
```

Comment:
We literally do not need this type in the `javaagent-core`. `javaagent-core` should contain only types that are used by multiple instrumentations. Which does not seem to be the case here. This type can be moved to undertow module. Also only  `SERVLET` value is used in the codebase.

I think I know your intentions with this type and overall approach, however it won't work as a generic solution across instrumentations. We should design an API that will tell you if payload or headers for a specific request were captured and it should work for both client and server instrumentations (clients have the same problem e.g. jax-rs client implementation is often times backed by apache HTTP client...).  The approach in this PR won't work bc it uses instrumentation context with specific types 3rd party library types e.g. `io.undertow.server.HttpServerExchange`.

You might want to look at https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/3e28b01e42311d755db17d698966d8f3bfaea1e3/instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/Instrumenter.java#L88 to see how to design this using the APIs that are available for all instrumentations.

Reply:
Definitely agreed, I'll move this type into a common library project shared by both `undertow` instrumentation projects. Over time, when we have more use-cases like this one, we can design a more general solution. This was definitely premature 🙂